### PR TITLE
chore: adding discv5 logs

### LINKED ISSUE #2841: chore: add logs when discovering and connecting to discv5 peers
### Background
To debug and further understand discv5's performance, as per the following [Discord thread](https://discord.com/channels/1110799176264056863/1252582625055473674), having logs documenting when a discv5 peer is discovered o connected to will help us get a clearer picture of discv5 by simulations made by the DHT team.

### Acceptance criteria
- [ ] a log is written when a discv5 peer is discovered with its details
- [ ] a log is written when we connect to a peer that was previously discovered via discv5

### PR DESCRIPTION
**Pull request by gabrielmer**
# Description

Adding logs when discv5 finds new peers and when the node connects to new peers. These logs will be under the compilation flag `DEBUG_DISCV5` so it is used for DST simulations.

It can be tested in the following image:

```
harbor.status.im/wakuorg/nwaku:discv5-debug-logs
```

Example logs
```
nwaku-1  | NTC 2024-06-25 12:07:43.869+00:00 Received discv5 node                       topics="waku discv5" tid=1 file=waku_discv5.nim:197 addrs=/ip4/195.201.172.178/tcp/30304/p2p/16Uiu2HAmKQAGsUAoUVMt6j7FLRWmou2z3a1WE6CGUiG1YKxWtq2u enr=enr:-LW4QK7dmsv43R2qQknllshhYx9IDEu1cD1t6Ao601ppx8oWYr_joTyfKXjoxQAST1iuGTXfs0ETtJ1fCrZef-GYHrsBgmlkgnY0gmlwhMPJrLKKbXVsdGlhZGRyc4CCcnOTAAEIAAAAAQACAAMABAAFAAYAB4lzZWNwMjU2azGhA2Q1UTaE09UZWli2y5AZFwE-3RGoGGvwrQ_9B5YAy_VWg3RjcIJ2YIN1ZHCCIy2Fd2FrdTIP capabilities="@[Relay, Store, Filter, Lightpush]" shards="(clusterId: 1, shardIds: @[0, 1, 2, 3, 4, 5, 6, 7])"
```

```
nwaku-1  | NTC 2024-06-25 12:08:53.816+00:00 established connections with found peers   topics="waku node peer_manager" tid=1 file=peer_manager.nim:670 peerIds="@[16Uiu2HAmB7Ur9HQqo3cWDPovRQjo57fxWWDaQx27WxSzDGhN4JKg, 16Uiu2HAmNTpGnyZ8W1BK2sXEmgSCNWiyDKgRU3NBR2DXST2HzxRU]" origin="@[Discv5, Discv5]"
```

CC @AlbertoSoutullo 

## Issue

related to #2810 
closes #2841

### CODE REVIEW

File: waku/node/peer_manager/peer_manager.nim [hunk 1]
```diff
@@ -649,18 +649,29 @@ proc connectToNodes*(
...
     if node.isOk():
       futConns.add(pm.connectRelay(node.value))
+      connectedPeers.add(node.value)
     else:
       error "Couldn't parse node info", error = node.error
 
   await allFutures(futConns)
-  let successfulConns = futConns.mapIt(it.read()).countIt(it == true)
+
+  # Filtering connectedPeers based on futConns
```
Comment (Ivansete-status) [inline waku/node/peer_manager/peer_manager.nim:663]:
```suggestion
  # Filtering succesful connectedPeers based on futConns
```

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100: 
I just added a minor nitpick comment
Very insightful PR and beautiful how you use the `zip` proc :partying_face:

File: waku/node/peer_manager/peer_manager.nim (hunk 1 above)
Comment (gabrielmer) [reply waku/node/peer_manager/peer_manager.nim]: Ooh good point! Updated it :)
Thanks!

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM, thank you!

**Pull request merged**